ui/src/components/RemoteFile.vue

```diff
@@ -0,0 +1,227 @@
+
+      list = list.map((line) => {
+        let html = "";
+        const record = JSON.parse(line);
+        let isAudit = false;
+        audit.forEach((field) => {
+          if (record[field]) {
+            isAudit = true;
+            html += mapper(field, record[field]);
+          }
+        });
+        if (isAudit) {
+          return (
+            html +
+            "\n" +
+            mapper("data", "\n" + JSON.stringify(record.data, null, 2))
+          );
+        } else {
+          return JSON.stringify(record.data, null, 2);
+        }
+      });
+    }
+    lines.value = list;
+    file.value = res;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+fetchFile();
+
+// Find line numbers with matching string values
+let matchedLines: number[] = [];
+function filteredLines() {
+  // find filter value in lines
+  const searchFor = filterValue.value.toLowerCase();
+  matchedLines = matchedLineIndices(lines.value, searchFor);
+
+  numberOfLines.value = matchedLines.length;
+  // FIXME: is this bad?
+  setTimeout(setFocusOnLine, 20, 0);
+}
+
+// Helper to highlight lines
+const isMatchedLine = (lineNo: number) => matchedLines.includes(lineNo);
+
+/**
+ * Scroll to one of the search results
+ *
+ * @param item index of element to set focus to.
+ */
+function setFocusOnLine(item: number) {
+  const elements = document.getElementsByClassName("text-danger");
+  if (elements.length > 0) {
+    if (item < 0) item = 0;
+    if (item >= elements.length) item = elements.length - 1;
+    if (item >= matchedLines.length) item = matchedLines.length - 1;
+    currentFocus.value = item;
+    elements[item].scrollIntoView();
+  }
+}
+
+function navigate(direction: string) {
+  if (direction === "first") {
+    currentFocus.value = 0;
+  } else if (direction === "prev") {
+    currentFocus.value -= 1;
+  } else if (direction === "next") {
+    currentFocus.value += 1;
+  } else if (direction === "last") {
+    currentFocus.value += matchedLines.length - 1;
+  }
+  setTimeout(setFocusOnLine, 20, currentFocus.value);
+}
+</script>
+
+<template>
+  <div v-if="file">
+    <div class="row">
+      <div class="col-sm-6">
+        <button class="btn btn-info" type="button" @click="fetchFile">
+          Reload @ server time {{ file.fetched }}
+        </button>
+        <a
+          class="btn btn-primary"
+          :href="'/insight/files/' + file.id + '/download'"
+          >Download</a
+        >
+      </div>
+    </div>
+    <div class="row">
+      <div class="col-sm-3">
+        <SearchBar id="searchbox" v-model="filterValue" />
+      </div>
+
+      <div class="col">
+        <div
+          class="btn-group"
+          role="group"
+          aria-label="navigation"
+          v-if="true || (filterValue && matchedLines.length > 0)"
+        >
+          <button
+            type="button"
+            class="btn btn-primary"
+            @click="navigate('first')"
+          >
+            |&lt
```

Comment:
I recommend you use an icon for this. I see we are using bootstrap icons (I would have prefered font awesome because more useful optioons) but then this might be an alternative?

https://icons.getbootstrap.com/icons/skip-backward-fill/
https://icons.getbootstrap.com/icons/skip-start-fill/
https://icons.getbootstrap.com/icons/skip-end-fill/
https://icons.getbootstrap.com/icons/skip-forward-fill/